src/mono/netcore/System.Private.CoreLib/src/System/ApplicationModel.Windows.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal static class ApplicationModel
+    {
+        // TODO: this is a readonly field in CoreCLR. Use PR to find the better 
+        // approach here since it seems wrong for this to differ between runtimes
+        internal static bool IsUap => throw new PlatformNotSupportedException();
```

Comment:
The CoreCLR version of `IsUap` is a `readonly static` property. For Mono I needed to insert a `PlatformNotSupportedException` which required an executable member. While this will compile I'm unsure if it meets our guidelines for how we swap out Mono / CoreCLR specific types. Appreciate any guidance here.

Reply:
That is fine. Internal APIs are not exposed via reference assemblies and whatever satisfies C# compiler works. If someone uses reflection to poke inside, he/she is in danger zone anyway.

Reply:
How does throw differ from false?